preliz/distributions/triangular.py

```diff
@@ -0,0 +1,223 @@
+            scale = upper - lower
+            c_ = (c - lower) / scale
+            Triangular(lower, c, upper).plot_pdf()
+
+    ========  ============================================================================
+    Support   :math:`x \in [lower, upper]`
+    Mean      :math:`\dfrac{lower + upper + c}{3}`
+    Variance  :math:`\dfrac{upper^2 + lower^2 +c^2 - lower*upper - lower*c - upper*c}{18}`
+    ========  ============================================================================
+
+    Parameters
+    ----------
+    lower : float
+        Lower limit.
+    c : float
+        Mode.
+    upper : float
+        Upper limit.
+    """
+
+    def __init__(self, lower=None, c=None, upper=None):
+        super().__init__()
+        self.support = (-np.inf, np.inf)
+        self._parametrization(lower, c, upper)
+
+    def _parametrization(self, lower=None, c=None, upper=None):
+        self.lower = lower
+        self.c = c
+        self.upper = upper
+        self.params = (self.lower, self.c, self.upper)
+        self.param_names = ("lower", "c", "upper")
+        self.params_support = ((-np.inf, np.inf), (-np.inf, np.inf), (-np.inf, np.inf))
+        if all_not_none(lower, c, upper):
+            self._update(lower, c, upper)
+
+    def _update(self, lower, c, upper):
+        self.lower = np.float64(lower)
+        self.c = np.float64(c)
+        self.upper = np.float64(upper)
+        self.support = (self.lower, self.upper)
+        self.params = (self.lower, self.c, self.upper)
+        self.is_frozen = True
+
+    def pdf(self, x):
+        """
+        Compute the probability density function (PDF) at a given point x.
+        """
+        x = np.asarray(x)
+        return np.exp(nb_logpdf(x, self.lower, self.c, self.upper))
+
+    def cdf(self, x):
+        """
+        Compute the cumulative distribution function (CDF) at a given point x.
+        """
+        x = np.asarray(x)
+        return nb_cdf(x, self.lower, self.c, self.upper)
+
+    def ppf(self, q):
+        """
+        Compute the percent point function (PPF) at a given probability q.
+        """
+        q = np.asarray(q)
+        return nb_ppf(q, self.lower, self.c, self.upper)
+
+    def logpdf(self, x):
+        """
+        Compute the log probability density function (log PDF) at a given point x.
+        """
+        return nb_logpdf(x, self.lower, self.c, self.upper)
+
+    def _neg_logpdf(self, x):
+        """
+        Compute the neg log_pdf sum for the array x.
+        """
+        return nb_neg_logpdf(x, self.lower, self.c, self.upper)
+
+    def entropy(self):
+        return nb_entropy(self.lower, self.upper)
+
+    def mean(self):
+        return (self.lower + self.c + self.upper) / 3
+
+    def median(self):
+        if self.c >= (self.lower + self.upper) / 2:
+            return self.lower + ((self.upper - self.lower) * (self.c - self.lower) / 2) ** 0.5
+        return self.upper - ((self.upper - self.lower) * (self.upper - self.c) / 2) ** 0.5
```

Comment:
Use np.where, so this becomes vectorized. Not sure how useful is to support moments for distributions with parameters being arrays. But we do for many distributions. At some point we should discuss if we want this.

Reply:
Sure, we can discuss on that
I have changed it to `np.where` as of now